### PR TITLE
CI Testing across SQL MySql and Postgres databases

### PR DESCRIPTION
**Pull request by deanmarcussen**
Related to https://github.com/OrchardCMS/OrchardCore/issues/6770

Implements some `Theory` attributes which skips tests when Connection strings are not supplied by the environment (CI or Local)

The idea is to run the integration tests against the SQL databases we support, rather than just Sqlite.

Primarily I'd like to do this because
- Sqlite doesn't support max column lengths, so we've had to fix a number of issues when users get to production. These could be better addressed by testing against a production database server.
- Table name lengths. Same issue - works on Sqlite, but MySql, and Postgres have smaller limits (here I've had to write a limited (read quick) table generator, because a guid is too long and MySql and Postgres both fail to create tables.)
- I'd like to do some optimisation of the indices on the index tables (including `DocumentId` and others) which are a lot easier to do and prove when one can run a test.

This might be controversial as @sebastienros had suggested to do this with the functional tests, rather than the integration tests.

Future extensions might include some test recipes (easy to keep in the test project without them leaking into nuget) to activate features like open id etc (again trying to test things like table lengths, and general setup), whereas the work @jptissot is doing with cypress would suit testing other pieces.

Would also be nice to spin up two servers and redis to test distributed... but we'll see.

I did have to downgrade xunit to get around this issue as well https://github.com/xunit/xunit/issues/1782

Now we'll see if the ci passes...

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -15,6 +15,11 @@ skip_commits:
     - mkdocs.yml
     - src/docs/**/*
 
+services:
+  - mssql2017
```
Comment (Skrypt) [inline appveyor.yml:19]: Nothing more recent? 2019?

Comment (Skrypt) [reply appveyor.yml:19]: https://www.appveyor.com/docs/services-databases/#sql-server-2019

**Comment by Skrypt** [comment]
So basically these unit tests won't execute on Travis because we won't supply a connectionstring.

Some thought : 

I'm not sure if we should still be testing for MacOS on Travis. It is slowing down the CI a lot.

File: appveyor.yml (hunk 1 above)
Comment (Skrypt) [reply appveyor.yml:19]: I can see why now.

File: test/OrchardCore.Tests/Apis/Context/Attributes/MySqlDataAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        public MySqlDataAttribute()
+        {
+            if (String.IsNullOrEmpty(Environment) && String.IsNullOrEmpty(Skip))
+            {
+                Skip = "MySql test skipped by environment";
+            }
+        }
+
+		public override IEnumerable<object?[]> GetData(MethodInfo testMethod)
```
Comment (Skrypt) [inline test/OrchardCore.Tests/Apis/Context/Attributes/MySqlDataAttribute.cs:31]: indentation is strange here and other DataAttribute inherited class.

File: test/OrchardCore.Tests/Apis/Context/Attributes/PostgreSqlDataAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+        public PostgreSqlDataAttribute()
+        {
+            if (String.IsNullOrEmpty(Environment) && String.IsNullOrEmpty(Skip))
+            {
+                Skip = "PostgreSql test skipped by environment";
+            }
+        }
+
+		public override IEnumerable<object?[]> GetData(MethodInfo testMethod)
```
Comment (Skrypt) [inline test/OrchardCore.Tests/Apis/Context/Attributes/PostgreSqlDataAttribute.cs:31]: indentation

Comment (deanmarcussen) [reply test/OrchardCore.Tests/Apis/Context/Attributes/PostgreSqlDataAttribute.cs]: Don't know why it's having such trouble with formatting here...

Anyway still trying to get sql server working

**Comment by deanmarcussen** [comment]
Ok, I'm happy this is passing reliably on appveyor.

I've now only tested each standard recipe against all 4 databases

(and removed one test which is covered by other tests, and creates unnecessary log entries during the test process)

More tests (I would like some) may come later on another pr

**Review by Skrypt — APPROVED** [review]
LGTM

**Pull request merged**